README
NAME

    CPAN::Perl::Releases - Mapping Perl releases on CPAN to the location of
    the tarballs

VERSION

    version 3.66

SYNOPSIS

      use CPAN::Perl::Releases qw[perl_tarballs];
    
      my $perl = '5.14.0';
    
      my $hashref = perl_tarballs( $perl );
    
      print "Location: ", $_, "\n" for values %{ $hashref };

DESCRIPTION

    CPAN::Perl::Releases is a module that contains the mappings of all perl
    releases that have been uploaded to CPAN to the authors/id/ path that
    the tarballs reside in.

    This is static data, but newer versions of this module will be made
    available as new releases of perl are uploaded to CPAN.

FUNCTIONS

    perl_tarballs

      Takes one parameter, a perl version to search for. Returns an hashref
      on success or undef otherwise.

      The returned hashref will have a key/value for each type of tarball.
      A key of tar.gz indicates the location of a gzipped tar file and
      tar.bz2 of a bzip2'd tar file. The values will be the relative path
      under authors/id/ on CPAN where the indicated tarball will be
      located.

        perl_tarballs( '5.14.0' );
      
        Returns a hashref like:
      
        {
          "tar.bz2" => "J/JE/JESSE/perl-5.14.0.tar.bz2",
          "tar.gz" => "J/JE/JESSE/perl-5.14.0.tar.gz"
        }

      Not all perl releases had tar.bz2, but only a tar.gz.

      Perl tarballs may also be compressed using xz and therefore have a
      tar.xz entry.

    perl_versions

      Returns the list of all the perl versions supported by the module in
      ascending order. TRIAL and RC will be lower than an actual release.

    perl_pumpkins

      Returns a sorted list of all PAUSE IDs of Perl pumpkins.

SEE ALSO

    http://www.cpan.org/src/5.0/

    http://search.cpan.org/faq.html#Is_there_a_API?

AUTHOR

    Chris Williams <chris@bingosnet.co.uk>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2018 by Chris Williams.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
==================================================
Changes from 2013-06-24 00:00:00 +0000 to present.
==================================================

-----------------------------------------
version 3.66 at 2018-06-23 08:49:18 +0000
-----------------------------------------

  Change: 71317c12d5690ebf283bf2cd8a8950df1ea5b5a4
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-06-23 09:49:18 +0000

    Updated for v5.28.0 

-----------------------------------------
version 3.64 at 2018-06-19 20:54:52 +0000
-----------------------------------------

  Change: 7e9a12d611aa91158c9c110bacdd85b51ee52cb4
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-06-19 21:54:52 +0000

    Updated for v5.28.0-RC4 

-----------------------------------------
version 3.62 at 2018-06-19 07:06:23 +0000
-----------------------------------------

  Change: 153ed06e1ee8e997b6bc193627be6f800f40b96f
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-06-19 08:06:23 +0000

    Updated for v5.28.0-RC3 

-----------------------------------------
version 3.60 at 2018-06-06 12:48:04 +0000
-----------------------------------------

  Change: 248af7199e59ead2ba910b9b2741d11356f69e9c
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-06-06 13:48:04 +0000

    Added v5.28.0-RC2 

-----------------------------------------
version 3.58 at 2018-05-21 13:21:57 +0000
-----------------------------------------

  Change: 9c2d1a38ad555ec8f3bf1afacb5da13ccd8387d2
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-05-21 14:21:57 +0000

    Updated for v5.28.0-RC1 

-----------------------------------------
version 3.56 at 2018-04-20 15:42:01 +0000
-----------------------------------------

  Change: 1f304574ea4bfd31f0052d1262e257043e673991
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-04-20 16:42:01 +0000

    Updated for v5.27.11 

-----------------------------------------
version 3.54 at 2018-04-14 15:53:18 +0000
-----------------------------------------

  Change: 24d6de304aeba48d7cbf82876ecd0b6a41c24919
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-04-14 16:53:18 +0000

    Updated for v5.24.4 & v5.26.2 

-----------------------------------------
version 3.52 at 2018-04-02 17:48:32 +0000
-----------------------------------------

  Change: 713ddd633fc582d440c3ea8e1b860888fa95a0a7
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-04-02 18:48:32 +0000

    Updated for v5.24.4 & v5.26.2 RC1 

-----------------------------------------
version 3.50 at 2018-03-20 21:20:56 +0000
-----------------------------------------

  Change: 89311141d4912c43ecc1a4f5e88a6dd51bf7009a
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-03-20 21:20:56 +0000

    Updated for v5.27.10 

-----------------------------------------
version 3.48 at 2018-02-20 21:16:37 +0000
-----------------------------------------

  Change: b4932c7dafb32915ca0830f9294b619eaf7a93f3
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-02-20 21:16:37 +0000

    Updated for v5.27.9 

-----------------------------------------
version 3.46 at 2018-01-20 11:07:29 +0000
-----------------------------------------

  Change: f56297606d2b5f0d4a73182943f1eb3bda46b0a4
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2018-01-20 11:07:29 +0000

    Updated for v5.27.8 

-----------------------------------------
version 3.44 at 2017-12-20 22:36:14 +0000
-----------------------------------------

  Change: 507f8355e9dad418d5dae235d0c0a3c8cc47760d
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-12-20 22:36:14 +0000

    Updated for v5.27.7 

-----------------------------------------
version 3.42 at 2017-11-20 23:24:42 +0000
-----------------------------------------

  Change: f9969c44f95f47f541ed64f21be32368d6566d12
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-11-20 23:24:42 +0000

    DAGOLDEN has removed .xz for v5.23.6 

  Change: ce509e7fc64975713d83bb05eb97cdf6ead2d871
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-11-20 23:18:35 +0000

    XSAWYER has removed all dev and RCs 

  Change: 2f02d02c0e215b3dcff2680013b4e94f17e05458
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-11-20 21:41:35 +0000

    Updated for v5.27.6 

  Change: 064a18f9b350102b2298a355d9be7408bc640bdf
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-11-20 21:33:20 +0000

    No bz2 tarballs after v5.27.5 

-----------------------------------------
version 3.40 at 2017-10-20 22:14:11 +0000
-----------------------------------------

  Change: daff9aedd0530e83e49b6e0445260747884ed31b
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-10-20 23:14:11 +0000

    Updated for v5.27.5 

-----------------------------------------
version 3.38 at 2017-09-23 08:44:14 +0000
-----------------------------------------

  Change: 128afc679b8c060d80ed1ab75c4648f6141e36d1
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-09-23 09:44:14 +0000

    Updated for v5.24.3 and v5.26.1 

-----------------------------------------
version 3.36 at 2017-09-20 22:00:13 +0000
-----------------------------------------

  Change: 7d16101a95e048802a847c74be48784cd9c1e2a5
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-09-20 23:00:13 +0000

    Updated for v5.27.4 

-----------------------------------------
version 3.34 at 2017-09-10 19:36:53 +0000
-----------------------------------------

  Change: a765d8e90e4e6133d3257fed3395f4c02c7ca19c
  Author: Chris 'BinGOs' Williams <chris@bingosnet.co.uk>
  Date : 2017-09-10 20:36:53 +0000

    Add 5.24.3-RC1 and 5.26.1-RC1 

-----------------------------------------
version 3.32 at 2017-08-21 22:34:58 +0000
-----------------------------------------


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.012.
use strict;
use warnings;

use 5.006000;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Mapping Perl releases on CPAN to the location of the tarballs",
  "AUTHOR" => "Chris Williams <chris\@bingosnet.co.uk>",
  "BUILD_REQUIRES" => {
    "Test::More" => "0.47"
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "CPAN-Perl-Releases",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006000",
  "NAME" => "CPAN::Perl::Releases",
  "PREREQ_PM" => {},
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => "0.47"
  },
  "VERSION" => "3.66",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Test::More" => "0.47"
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


lib/CPAN/Perl/Releases.pm
package CPAN::Perl::Releases;
$CPAN::Perl::Releases::VERSION = '3.66';
#ABSTRACT: Mapping Perl releases on CPAN to the location of the tarballs

use strict;
use warnings;
use vars qw[@ISA @EXPORT_OK];

use Exporter;

@ISA       = qw(Exporter);
@EXPORT_OK = qw(perl_tarballs perl_versions perl_pumpkins);

# Data gathered from using findlinks.pl script in this dists tools/
# directory, run over the src/5.0 of a local CPAN mirror.
our $cache = { };
our $data =
{
"5.004" => { id => 'CHIPS' },
"5.004_01" => { id => 'TIMB' },
"5.004_02" => { id => 'TIMB' },
"5.004_03" => { id => 'TIMB' },
"5.004_04" => { id => 'TIMB' },
"5.004_05" => { id => 'CHIPS' },
"5.005" => { id => 'GSAR' },
"5.005_01" => { id => 'GSAR' },
"5.005_02" => { id => 'GSAR' },
"5.005_03" => { id => 'GBARR' },
"5.005_04" => { id => 'LBROCARD' },
"5.6.0" => { id => 'GSAR' },
"5.6.1-TRIAL1" => { id => 'GSAR' },
"5.6.1-TRIAL2" => { id => 'GSAR' },
"5.6.1-TRIAL3" => { id => 'GSAR' },
"5.6.1" => { id => 'GSAR' },
"5.6.2" => { id => 'RGARCIA' },
"5.7.0" => { id => 'JHI' },
"5.7.2" => { id => 'JHI' },
"5.7.3" => { id => 'JHI' },
"5.8.0" => { id => 'JHI' },
"5.8.1" => { id => 'JHI' },
"5.8.2" => { id => 'NWCLARK' },
"5.8.3" => { id => 'NWCLARK' },
"5.8.4" => { id => 'NWCLARK' },
"5.8.5" => { id => 'NWCLARK' },
"5.8.6" => { id => 'NWCLARK' },
"5.8.7" => { id => 'NWCLARK' },
"5.8.8" => { id => 'NWCLARK' },
"5.8.9" => { id => 'NWCLARK' },
"5.9.0" => { id => 'HVDS' },
"5.9.1" => { id => 'RGARCIA' },
"5.9.2" => { id => 'RGARCIA' },
"5.9.3" => { id => 'RGARCIA' },
"5.9.4" => { id => 'RGARCIA' },
"5.9.5" => { id => 'RGARCIA' },
"5.10.0" => { id => 'RGARCIA' },
"5.10.1" => { id => 'DAPM' },
"5.11.0" => { id => 'JESSE' },
"5.11.1" => { id => 'JESSE' },
"5.11.2" => { id => 'LBROCARD' },
"5.11.3" => { id => 'JESSE' },
"5.11.4" => { id => 'RJBS' },
"5.11.5" => { id => 'SHAY' },
"5.12.0" => { id => 'JESSE' },
"5.12.1" => { id => 'JESSE' },
"5.12.2" => { id => 'JESSE' },
"5.12.3" => { id => 'RJBS' },
"5.12.4" => { id => 'LBROCARD' },
"5.12.5" => { id => 'DOM' },
"5.13.0" => { id => 'LBROCARD' },
"5.13.1" => { id => 'RJBS' },
"5.13.2" => { id => 'MSTROUT' },
"5.13.3" => { id => 'DAGOLDEN' },
"5.13.4" => { id => 'FLORA' },
"5.13.5" => { id => 'SHAY' },
"5.13.6" => { id => 'MIYAGAWA' },
"5.13.7" => { id => 'BINGOS' },
"5.13.8" => { id => 'ZEFRAM' },
"5.13.9" => { id => 'JESSE' },
"5.13.10" => { id => 'AVAR' },
"5.13.11" => { id => 'FLORA' },
"5.14.0" => { id => 'JESSE' },
"5.14.1" => { id => 'JESSE' },
"5.14.2-RC1" => { id => 'FLORA' },
"5.14.2" => { id => 'FLORA' },
"5.14.3" => { id => 'DOM' },
"5.14.4-RC1" => { id => 'DAPM' },
"5.14.4-RC2" => { id => 'DAPM' },
"5.14.4" => { id => 'DAPM' },
"5.15.0" => { id => 'DAGOLDEN' },
"5.15.1" => { id => 'ZEFRAM' },
"5.15.2" => { id => 'RJBS' },
"5.15.3" => { id => 'STEVAN' },
"5.15.4" => { id => 'FLORA' },
"5.15.5" => { id => 'SHAY' },
"5.15.6" => { id => 'DROLSKY' },
"5.15.7" => { id => 'BINGOS' },
"5.15.8" => { id => 'CORION' },
"5.15.9" => { id => 'ABIGAIL' },
"5.16.0" => { id => 'RJBS' },
"5.16.1" => { id => 'RJBS' },
"5.16.2" => { id => 'RJBS' },
"5.16.3" => { id => 'RJBS' },
"5.17.0" => { id => 'ZEFRAM' },
"5.17.1" => { id => 'DOY' },
"5.17.2" => { id => 'TONYC' },
"5.17.3" => { id => 'SHAY' },
"5.17.4" => { id => 'FLORA' },
"5.17.5" => { id => 'FLORA' },
"5.17.6" => { id => 'RJBS' },
"5.17.7" => { id => 'DROLSKY' },
"5.17.8" => { id => 'ARC' },
"5.17.9" => { id => 'BINGOS' },
"5.17.10" => { id => 'CORION' },
"5.17.11" => { id => 'RJBS' },
"5.18.0" => { id => 'RJBS' },
"5.18.1" => { id => 'RJBS' },
"5.19.0" => { id => 'RJBS' },
"5.19.1" => { id => 'DAGOLDEN' },
"5.19.2" => { id => 'ARISTOTLE' },
"5.19.3" => { id => 'SHAY' },
"5.19.4" => { id => 'SHAY' },
"5.19.5" => { id => 'SHAY' },
"5.19.6" => { id => 'BINGOS' },
"5.19.7" => { id => 'ABIGAIL' },
"5.18.2" => { id => 'RJBS' },
"5.19.8" => { id => 'RJBS' },
"5.19.9" => { id => 'TONYC' },
"5.19.10" => { id => 'ARC' },
"5.19.11" => { id => 'SHAY' },
"5.20.0" => { id => 'RJBS' },
"5.21.0" => { id => 'RJBS' },
"5.21.1" => { id => 'WOLFSAGE' },
"5.21.2" => { id => 'ABIGAIL' },
"5.21.3" => { id => 'PCM' },
"5.20.1-RC1" => { id => 'SHAY' },
"5.20.1-RC2" => { id => 'SHAY' },
"5.20.1" => { id => 'SHAY' },
"5.21.4" => { id => 'SHAY' },
"5.18.3" => { id => 'RJBS' },
"5.18.4" => { id => 'RJBS' },
"5.21.5" => { id => 'ABIGAIL' },
"5.21.6" => { id => 'BINGOS' },
"5.21.7" => { id => 'CORION' },
"5.21.8" => { id => 'WOLFSAGE' },
"5.20.2-RC1" => { id => 'SHAY' },
"5.20.2" => { id => 'SHAY' },
"5.21.10" => { id => 'SHAY' },
"5.21.11" => { id => 'SHAY' },
"5.22.0" => { id => 'RJBS' },
"5.23.0" => { id => 'RJBS' },
"5.23.1" => { id => 'WOLFSAGE' },
"5.23.2" => { id => 'WOLFSAGE' },
"5.20.3-RC1" => { id => 'SHAY' },
"5.20.3-RC2" => { id => 'SHAY' },
"5.20.3" => { id => 'SHAY' },
"5.23.3" => { id => 'PCM' },
"5.23.4" => { id => 'SHAY' },
"5.22.1-RC1" => { id => 'SHAY' },
"5.22.1-RC2" => { id => 'SHAY' },
"5.23.5" => { id => 'ABIGAIL' },
"5.22.1-RC3" => { id => 'SHAY' },
"5.22.1-RC4" => { id => 'SHAY' },
"5.22.1" => { id => 'SHAY' },
"5.23.6" => { id => 'DAGOLDEN', noxz => 1 },
"5.23.7" => { id => 'STEVAN' },
"5.23.9" => { id => 'ABIGAIL' },
"5.22.2-RC1" => { id => 'SHAY' },
"5.24.0-RC1" => { id => 'RJBS' },
"5.24.0-RC2" => { id => 'RJBS' },
"5.24.0-RC3" => { id => 'RJBS' },
"5.22.2" => { id => 'SHAY' },
"5.24.0-RC4" => { id => 'RJBS' },
"5.24.0-RC5" => { id => 'RJBS' },
"5.24.0" => { id => 'RJBS' },
"5.25.0" => { id => 'RJBS' },
"5.25.2" => { id => 'WOLFSAGE' },
"5.22.3-RC1" => { id => 'SHAY' },
"5.24.1-RC1" => { id => 'SHAY' },
"5.25.3" => { id => 'SHAY' },
"5.22.3-RC2" => { id => 'SHAY' },
"5.24.1-RC2" => { id => 'SHAY' },
"5.22.3-RC3" => { id => 'SHAY' },
"5.24.1-RC3" => { id => 'SHAY' },
"5.25.4" => { id => 'BINGOS' },
"5.25.5" => { id => 'STEVAN' },
"5.22.3-RC4" => { id => 'SHAY' },
"5.24.1-RC4" => { id => 'SHAY' },
"5.25.6" => { id => 'ARC' },
"5.25.7" => { id => 'EXODIST' },
"5.22.3-RC5" => { id => 'SHAY' },
"5.24.1-RC5" => { id => 'SHAY' },
"5.22.3" => { id => 'SHAY' },
"5.24.1" => { id => 'SHAY' },
"5.25.9" => { id => 'ABIGAIL' },
"5.25.10" => { id => 'RENEEB' },
"5.26.0" => { id => 'XSAWYERX' },
"5.27.1" => { id => 'EHERMAN' },
"5.22.4-RC1" => { id => 'SHAY' },
"5.24.2-RC1" => { id => 'SHAY' },
"5.22.4" => { id => 'SHAY' },
"5.24.2" => { id => 'SHAY' },
"5.27.2" => { id => 'ARC' },
"5.27.3" => { id => 'WOLFSAGE' },
"5.24.3-RC1" => { id => 'SHAY' },
"5.26.1-RC1" => { id => 'SHAY' },
"5.27.4" => { id => 'GENEHACK' },
"5.24.3" => { id => 'SHAY' },
"5.26.1" => { id => 'SHAY' },
"5.27.5" => { id => 'SHAY' },
"5.27.6" => { id => 'ETHER' },
"5.27.7" => { id => 'BINGOS' },
"5.27.8" => { id => 'ABIGAIL' },
"5.27.9" => { id => 'RENEEB' },
"5.27.10" => { id => 'TODDR' },
"5.24.4-RC1" => { id => 'SHAY' },
"5.26.2-RC1" => { id => 'SHAY' },
"5.24.4" => { id => 'SHAY' },
"5.26.2" => { id => 'SHAY' },
"5.27.11" => { id => 'XSAWYERX' },
"5.28.0-RC1" => { id => 'XSAWYERX' },
"5.28.0-RC2" => { id => 'XSAWYERX' },
"5.28.0-RC3" => { id => 'XSAWYERX' },
"5.28.0-RC4" => { id => 'XSAWYERX' },
"5.28.0" => { id => 'XSAWYERX' },
};

sub perl_tarballs {
  my $vers = shift;
  $vers = shift if eval { $vers->isa(__PACKAGE__) };
  return unless exists $data->{ $vers };
  if ( exists $cache->{ $vers } ) {
    return { %{ $cache->{ $vers } } };
  }
  my $pumpkin = $data->{ $vers }->{id};
  my $path = join '/', substr( $pumpkin, 0, 1 ), substr( $pumpkin, 0, 2 ), $pumpkin;
  my $sep = ( $vers =~ m!^5\.0! ? '' : '-' );
  my $perl = join $sep, 'perl', $vers;
  my $onlygz = 1 if $vers =~ m!(?-xism:5.(?:00(?:4(?:_0[12345])?|5(?:_0[1234])?|3_07)|1(?:0.0(?:-RC[12])?|6.0-RC0)|6.(?:[02]|1(?:-TRIAL[123])?)|9.[12345]|7.[0123]|8.[01]))! || $data->{ $vers }->{onlygz};
  my $onlybz2 = 1 if $data->{ $vers }->{onlybz2};
  my $noxz = 1 if $data->{ $vers }->{noxz};
  my $lvers;
  {
    my $tvers = $vers;
    $tvers =~ s!\-?(TRIAL|RC)\d*!!g;
    $tvers =~ s!_!.!g;
    my @parts = split m!\.!, $tvers;
    push @parts, 0 if scalar @parts < 3;
    $lvers = sprintf("%d.%03d%03d",@parts);
  }
  my $foo = { };
  $foo->{'tar.gz'} = "$path/$perl.tar.gz" unless $onlybz2;
  $foo->{'tar.bz2'} = "$path/$perl.tar.bz2" unless $onlygz || $lvers > 5.027005;
  $foo->{'tar.xz'} = "$path/$perl.tar.xz" if $lvers > 5.021005 && !$noxz;
  $cache->{ $vers } = $foo;
  return { %$foo };
}

sub perl_versions {
    return sort _by_version keys %$data;
}


sub _by_version {
  my %v = map {
    my @v = split(qr/[-._]0*/, $_);
    $v[2] ||= 0;
    $v[3] ||= 'Z';
    ($_ => sprintf '%d.%03d%03d-%s', @v)
  } $a, $b;
  $v{$a} cmp $v{$b};
}

sub perl_pumpkins {
    my %pumps = map { ( $data->{$_}->{id} => 1 ) } keys %$data;
    return sort keys %pumps;
}

q|Acme::Why::Did::I::Not::Read::The::Fecking::Memo|;

__END__

=pod

=encoding UTF-8

=head1 NAME

CPAN::Perl::Releases - Mapping Perl releases on CPAN to the location of the tarballs

=head1 VERSION

version 3.66

=head1 SYNOPSIS

  use CPAN::Perl::Releases qw[perl_tarballs];

  my $perl = '5.14.0';

  my $hashref = perl_tarballs( $perl );

  print "Location: ", $_, "\n" for values %{ $hashref };

=head1 DESCRIPTION

CPAN::Perl::Releases is a module that contains the mappings of all C<perl> releases that have been uploaded to CPAN to the
C<authors/id/> path that the tarballs reside in.

This is static data, but newer versions of this module will be made available as new releases of C<perl> are uploaded to CPAN.

=head1 FUNCTIONS

=over

=item C<perl_tarballs>

Takes one parameter, a C<perl> version to search for. Returns an hashref on success or C<undef> otherwise.

The returned hashref will have a key/value for each type of tarball. A key of C<tar.gz> indicates the location
of a gzipped tar file and C<tar.bz2> of a bzip2'd tar file. The values will be the relative path under C<authors/id/>
on CPAN where the indicated tarball will be located.

  perl_tarballs( '5.14.0' );

  Returns a hashref like:

  {
    "tar.bz2" => "J/JE/JESSE/perl-5.14.0.tar.bz2",
    "tar.gz" => "J/JE/JESSE/perl-5.14.0.tar.gz"
  }

Not all C<perl> releases had C<tar.bz2>, but only a C<tar.gz>.

Perl tarballs may also be compressed using C<xz> and therefore have a C<tar.xz> entry.

=item C<perl_versions>

Returns the list of all the perl versions supported by the module in ascending order. C<TRIAL> and C<RC> will be lower
than an actual release.

=item C<perl_pumpkins>

Returns a sorted list of all PAUSE IDs of Perl pumpkins.

=back

=head1 SEE ALSO

L<http://www.cpan.org/src/5.0/>

L<http://search.cpan.org/faq.html#Is_there_a_API?>

=head1 AUTHOR

Chris Williams <chris@bingosnet.co.uk>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2018 by Chris Williams.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut
